internal/backends/winit/renderer/skia.rs

```diff
@@ -199,11 +199,9 @@ impl i_slint_core::renderer::Renderer for SkiaRenderer {
             i_slint_core::items::TextVerticalAlignment::Bottom => max_height - layout.height(),
         };
 
-        layout
-            .get_glyph_position_at_coordinate((pos.x, pos.y - layout_top_y))
-            .position
-            .try_into()
-            .unwrap_or_default()
+        let glyph_index =
+            layout.get_glyph_position_at_coordinate((pos.x, pos.y - layout_top_y)).position;
+        string.char_indices().nth(glyph_index as usize).unwrap_or((string.len(), '\0')).0
```

Comment:
Well spotted, this was indeed wrong. I assumed that because `SkString` is utf-8 encoded, the returned value is a byte offset. Looking at the code in `TextLine::getGlyphPositionAtCoordinate` it appears that this is also not the glyph index but the index in the string as if it were utf-16 encoded. Perhaps it's that way because of the JavaScript bindings. So strictly speaking `chars().nth()` isn't correct either, but it's most certainly an improvement.

```suggestion
        let utf16_index =
            layout.get_glyph_position_at_coordinate((pos.x, pos.y - layout_top_y)).position;
        // TODO: this breaks with surrogate pairs.
        string.char_indices().nth(utf16_index as usize).unwrap_or((string.len(), '\0')).0
```

Reply:
wait, utf16?   this is not the utf16 index. it's the glyph index.  we'd need to use some other function to get the utf16 index.